### PR TITLE
Check for merge conflicts before creating SRPM

### PR DESCRIPTION
**Pull request by mfocko**
In addition to merging pull request changes into base branch before
running SRPM build, check for potential merge conflicts and set neutral
status, if detected.

Related to packit/packit#1344

Merge after packit/packit#1344 (To ensure image is rebuilt with the latest packit)

---

Packit Service now runs Copr and Koji builds and following tests on Testing Farm for pull requests on the code that would be a result of merging into the target branch. In case merge conflicts occur during preparation of SRPM, you can find more info in the SRPM logs.

### CODE REVIEW

**Comment by mfocko** [comment]
Bumping priority of refactoring fixtures…

File: packit_service/worker/build/build_helper.py
Comment (TomasTomecek) [inline packit_service/worker/build/build_helper.py:353]: This can still fail due to a race condition - time of check & time of use. So we should prepared here.

File: packit_service/worker/build/build_helper.py
Comment (lachmanfrantisek) [inline packit_service/worker/build/build_helper.py:333]: This can be a bit unclear for someone. But I am not strictly against it. (I see how it makes the code shorter.)

**Review by TomasTomecek — APPROVED** [review]
since most of the PRs can be merged when they are created and only decay over time, I'd try this :)

File: packit_service/worker/build/build_helper.py
Comment (lachmanfrantisek) [inline packit_service/worker/build/build_helper.py:341]: Can we believe this after all?

**Review by lachmanfrantisek — APPROVED** [review]

File: packit_service/worker/build/build_helper.py
Comment (mfocko) [reply packit_service/worker/build/build_helper.py:341]: I think the main concern with the test merge commit was not knowing what it was merged with. Since that is handled with the logging, I guess it shouldn't be a problem.

I have decided to switch to catching the exception, since it's the most reliable imo.

File: packit_service/worker/build/build_helper.py
Comment (mfocko) [reply packit_service/worker/build/build_helper.py:353]: you mean race condition between maintainers/CI and the „state“ from PR?

**Comment by mfocko** [comment]
_packit/packit#1344 became a hard dependency… introduced `PackitMergeException`_

**Review by lachmanfrantisek — APPROVED** [review]
Looks good to me, thanks!

**Comment by mfocko** [comment]
recheck

**Comment by mfocko** [comment]
recheck

**Comment by mfocko** [comment]
_tests are failing because of cached test image with version of packit that doesn't include `PackitMergeException`, merging manually_

**Pull request merged**
